src/citrine/builders/design_spaces.py

```diff
@@ -87,36 +126,57 @@ def enumerate_simple_mixture_cartesian_product(
     form_ds = form_ds[form_ds[balance_component] >= balance_range[0]]
     form_ds = form_ds[form_ds[balance_component] <= balance_range[1]]
     data = form_ds.to_dict('records')
-
-    return data
+    design_space = EnumeratedDesignSpace(name=name,
+                                         description=description,
+                                         descriptors=descriptors,
+                                         data=data)
+    return design_space
 
 
 def cartesian_join_design_spaces(
-    data_list: Sequence[Sequence[Mapping[str, Any]]],
-) -> Sequence[Mapping[str, Any]]:
-    ''' Given multiple input design spaces (lists of candidates), create a
-    set of candidates that is the Cartesian product of candidates from each
+    design_space_list: List[EnumeratedDesignSpace],
+    name: str = 'Joined enumerated design space',
+    description: str = '',
+) -> EnumeratedDesignSpace:
+    """[ALPHA] Given a list of multiple input EnumeratedDesignSpaces, create a new
+    EnumeratedDesignSpace that is the Cartesian product of candidates from each
     input design space.
     
-    ultimate length will be len(data_list[0])*len(data_list[2])*...
+    Ultimate length will be the product of the number of candidates in each
+    individual design space
     
     Parameters
     ----------
-    data_list: Sequence[Sequence[Mapping[str, Any]]]
-        A list of [lists of candidates] (data for individual design subspaces)
-    '''
+    design_space_list: List[EnumeratedDesignSpace]
+        A list of EnumeratedDesignSpaces
+    """
     
-    assert 'join_key' not in [data[0].keys() for data in data_list], \
+    assert 'join_key' not in [ds.data[0].keys() for ds in design_space_list], \
         '"join_key" cannot be a descriptor key when using this function'
     
+    # Convert data fields of EDS into DataFrames to prep for join
+    data_list = [ds.data for ds in design_space_list]
     ds_list = [pd.DataFrame(data) for data in data_list]
     
+    # Set a dummy column to do the join
     for df in ds_list:
         df['join_key'] = 0
     
+    # Perform the join
     df_join = ds_list[0]
     for df in ds_list[1:]:
         df_join = pd.merge(df_join, df, on='join_key')
+        
```

Comment:
@maxhutch this is the logic we would need to replace from pandas. Very simple outer join...